include/stl2/view/subrange.hpp

```diff
@@ -23,6 +23,17 @@
 #include <stl2/view/view_interface.hpp>
 
 STL2_OPEN_NAMESPACE {
+	template<class From, class To>
+	STL2_CONCEPT _ConvertibleToNotSlicing =
+		ConvertibleTo<From, To> &&
+		// A conversion is a slicing conversion if the source and the destination
+		// are both pointers, and if the pointed-to types differ after removing
+		// cv qualifiers.
+		!(_Is<__uncvref<From>, std::is_pointer> &&
```

Comment:
We need a decay here for `From` and not just `__uncvref`. Consider `pair<Derived(&)[10], some_sentinel>`  used with `_PairLikeConvertibleTo`.

Reply:
I feel I should point out that none of this process is making me any more fond of the conversion from pairs. Tell me again why we want this conversion to work?

Reply:
I worry also about conversion from things that are _both_ ranges and pair-like, which would be ambiguous today. Yeah, I dunno. I can't think of a really compelling use case right now, except interoperability with old `std::` algorithms like `equal_range` that returns a `std::pair` that represents a range.

Reply:
>  interoperability with old `std::` algorithms like `equal_range` that returns a `std::pair` that represents a range.

I don't know about you, but I've had to explain to people many times why we "don't just make `pair<iterator, iterator>` be a range." It seems like a bad idea to take a step backward and treat them like they denote ranges when we've been explaining to people that an arbitrary pair of iterators is just two iterators with no more semantic meaning. 

(And yes, this is a separate issue.)

Reply:
> I worry also about conversion from things that are both ranges and pair-like, which would be ambiguous today.

We have a `!Range` constraint in the *`pairlike-convertible-XXX`* concepts to disambiguate them.